### PR TITLE
Adding the scaffolding for the GHA

### PR DESCRIPTION
**Pull request by stack72**

### CODE REVIEW

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -0,0 +1,557 @@
...
+      - name: Update path
+        run: echo ::add-path::${{ github.workspace }}/bin
+      - name: Install Python deps
+        run: |-
+          pip3 install virtualenv==20.0.23
+          pip3 install pipenv
+      - name: Install dependencies
+        run: make install_${{ matrix.language}}_sdk
+      - run: mkdir -p "$HOME/.kube/"
+      - name: Download Kueconfig
```
Comment (lblackstone) [inline .github/workflows/master.yml:315]:
```suggestion
      - name: Download Kubeconfig
```

File: Makefile [hunk 2]
```diff
@@ -9,102 +8,85 @@ NUGET_PKG_NAME   := Pulumi.Kubernetes
...
 		dotnet build /p:Version=${DOTNET_VERSION}
 
-go_sdk:: k8sgen $(SCHEMA_FILE)
-	$(CODEGEN) -version=${VERSION} go $(SCHEMA_FILE) $(CURDIR)
+go_sdk::
+	$(WORKING_DIR)/bin/$(CODEGEN) -version=${VERSION} go $(SCHEMA_FILE) $(CURDIR)
 
-nodejs_sdk:: k8sgen $(SCHEMA_FILE)
-	$(CODEGEN) -version=${VERSION} nodejs $(SCHEMA_FILE) $(CURDIR)
+nodejs_sdk:: VERSION := $(shell pulumictl get version --language javascript)
```
Comment (lblackstone) [inline Makefile:55]: Is this intentionally masking the `VERSION` set in line 11?

File: Makefile [hunk 3]
```diff
@@ -9,102 +8,85 @@ NUGET_PKG_NAME   := Pulumi.Kubernetes
...
+nodejs_sdk::
+	$(WORKING_DIR)/bin/$(CODEGEN) -version=${VERSION} nodejs $(SCHEMA_FILE) $(CURDIR)
 	cd ${PACKDIR}/nodejs/ && \
 		yarn install && \
 		yarn run tsc
 	cp README.md LICENSE ${PACKDIR}/nodejs/package.json ${PACKDIR}/nodejs/yarn.lock ${PACKDIR}/nodejs/bin/
 	sed -i.bak 's/$${VERSION}/$(VERSION)/g' ${PACKDIR}/nodejs/bin/package.json
 
-python_sdk:: k8sgen $(SCHEMA_FILE)
+nodejs_sdk:: PYPI_VERSION := $(shell pulumictl get version --language python)
```
Comment (lblackstone) [inline Makefile:64]: I think this should be `python_sdk`
```suggestion
python_sdk:: PYPI_VERSION := $(shell pulumictl get version --language python)
```

File: Makefile [hunk 4]
```diff
@@ -127,26 +108,13 @@ test_all::
...
-.PHONY: travis_cron travis_push travis_pull_request travis_api
-travis_cron: all
-travis_push: only_build check_clean_worktree publish_tgz only_test publish_packages
-travis_pull_request: only_build check_clean_worktree only_test
-travis_api: all
+install_dotnet_sdk::
+	mkdir -p $(WORKING_DIR)/nuget
+	find . -name '*.nupkg' -print -exec cp -p {} ${WORKING_DIR}/nuget \;
+
+install_python_sdk::
```
Comment (lblackstone) [inline Makefile:115]: Maybe add a comment that this is intentionally blank?

**Review by lblackstone — CHANGES_REQUESTED** [review]
This is looking great overall! 🎉

File: Makefile (hunk 2 above)
Comment (stack72) [reply Makefile:55]: yeah - the pulumictl cli app creates language specific (i.e. semver compliant) versions so we set the top one just as a base and then we override it for each specific language

**Review by lblackstone — APPROVED** [review]

File: .github/workflows/pull-request.yml [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+      - name: Install Kubectl
+        run: |
+          curl -LO https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl
+          chmod +x ./kubectl
+          sudo mv kubectl /usr/local/bin
+      - name: Install Kubectl
+        run: |
+          curl -LO https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl
+          chmod +x ./kubectl
+          sudo mv kubectl /usr/local/bin
```
Comment (lblackstone) [inline .github/workflows/pull-request.yml:295]:
```suggestion
```
Duplicated step here

**Pull request merged**